woocommerce/google-listings-and-ads
# Fetch WP Assets from specific post.

### Changes proposed in this Pull Request:



Closes part of https://github.com/woocommerce/google-listings-and-ads/issues/1780

This PR adds a new endpoint `assets/suggestions` that returns a list of assets for a specific post (page/product/post/etc),  with the below mapping:

Google text asset | Existing post/taxonomy meta
-- | --
Final URL | Permalink
(1) Headline Var1 Max. 15 chars | Post/taxonomy title
(2) Headline Var2 Max. 30 chars | Post/taxonomy title
(3) Long headline Max. 90 chars | Site title: Post/taxonomy title
(4) Short description Max. 60 chars | Post excerpt/taxonomy description
(5) Description Max. 90 chars | Post excerpt/taxonomy description, tagline
(5) Business name Max. 25 chars | Site title
(7) Call-to-action | –
(8) Display URL path | Slug


See original table: pcTzPl-12o-p2#comment-1849

- The Headlines (1 & 2) are grouped as one headline, as the Ads API only use one type which is [HEADLINE](https://github.com/googleads/google-ads-php/blob/151bab9bd37d46543bbeb2c3b0caedf91cb91b6b/src/Google/Ads/GoogleAds/V11/Enums/AssetFieldTypeEnum/AssetFieldType.php#L177).
- The Descriptions (4 & 5) are grouped as one description, as the Ads API only use one type which is [DESCRIPTION](https://github.com/googleads/google-ads-php/blob/151bab9bd37d46543bbeb2c3b0caedf91cb91b6b/src/Google/Ads/GoogleAds/V11/Enums/AssetFieldTypeEnum/AssetFieldType.php#L178).
- When fetching suggestions from WP/WC the field `call_to_action` will be `null` as it is not possible to offer a suggestion, however when we suggest assets used in other campaigns using the Ads API, the field will be filled. 

In addition to the mapping table, the endpoint returns the image assets linked to the post and the website logo.

At the moment we have three types of images:
- Square Marketing images.
- Marketing Images (landscape).
- Logo

The endpoint will return the same images for Square and Landascape images until we decide how to to resize/format the images. See comment: pcTzPl-12o-p2#comment-2090

The images that will return for a post are a combination of:
- Images attached to the post.
- Gallery images
- In case that is a product, images from the product gallery.
- If the page is the Shop page, it will return images of published products. 

### API Response:

```
{
    "headline": [
        "Sample Page"
    ],
    "long_headline": [
        "Wordpress: Sample Page"
    ],
    "description": [
        "I sell clothing!"
    ],
    "logo": [
        "http://localhost:8082/wp-content/uploads/2022/11/cropped-cropped-ocean-3605547_960_720-150x150.jpeg"
    ],
    "final_url": "http://localhost:8082/sample-page/",
    "business_name": "Wordpress",
    "display_url_path": [
        "sample-page"
    ],
    "square_marketing_images": [
        "http://localhost:8082/wp-content/uploads/2022/11/ocean-3605547_960_720-150x150.jpeg",
        ........
    ],
    "marketing_images": [
        "http://localhost:8082/wp-content/uploads/2022/11/Terminator-2-judgement-day-v3.jpeg",
        ..........
    ],
    call_to_action: null
}
```

### Detailed test instructions:


1. Chose any post created in your website. A post can be a product,page,post or any other custom post_type.
2. Get the ID of that post.
3. Make a request to: `assets/suggestions?id=YOUR_ID&type=post`
4. Check that the response matches the mapping table.


### Additional details:
New PRs will include fetching assets for a specific term/category and assets from other campaigns.


### Changelog entry
